### ISSUE TITLE
Why is the finish event fired before everything is done?

### ISSUE THREAD

**Issue by callumacrae** [juliangruber/multipipe#13]
```
var combined = multipipe(stream1, stream2)
```

The current behaviour seems to be that when `stream1` fires the "finish" event, it is then fired on "combined", and when `stream2` fires the "finish" event, it is ignored. Why not wait for both to finish? In it's current form, it's not very useful as it fires before everything is finished.

Proposed: `stream1` fires the "finish" event, and it is ignored. Then, `stream2` fires the "finish" event. Everything in the stream is now finished, so the "finish" event is fired on `combined`.

If you're happy with this behaviour change, I'd be happy to implement it, because it's causing a pretty bad bug here—the stream is ending early, causing the process to quit before everything is complete.

---

Also, why are you using "finish" instead of [end](https://nodejs.org/api/stream.html#stream_event_end)?

**Comment by juliangruber** [comment]
sounds like an issue with duplexer2, since multipart has nothing to do with those events really. can you open an issue there?

**Comment by callumacrae** [comment]
Having had another look, it looks like I just suck at streams. I *think* the finish event is fired multiple times, and I should have been using the end event instead.

Code before:

```
task('whatever', function (done) {
    var stream = multipipe(stream1, stream2);
    stream.on('finish', function () {
        done(); // exits the process
    });
});
```

Code now:

```
task('whatever', function (done) {
    var stream = multipipe(stream1, stream2);
    stream.resume();
    stream.on('end', function () {
        done(); // exits the process
    });
});
```

tl;dr: end != finish
